File: Report/projects/eventBus/remote/RemoteActor.scala
package it.infinity.model

import akka.actor.Actor
import akka.actor.ActorRef

class RemoteActor extends Actor {
	private val messageBus: MessageBus = new MessageBus();
	
	def receive = {
		case Subscribe(actorRef, n) => messageBus.subscribe(actorRef, n);
		case Publish(message) => messageBus.publish(message);
	}
}

case class Subscribe(val actorRef: ActorRef, val n: Int) { }
case class Publish(val content: String) { }

File: Projects/JavaAkkaServer/src/it/infinity/actor/RemoteActor.java
package it.infinity.actor;

import akka.actor.AbstractActor;
import akka.actor.Props;
import it.infinity.controller.Manager;
import it.infinity.message.AddEventMessage;
import it.infinity.message.GetEventsMessage;

public class RemoteActor extends AbstractActor {
	private Manager manager;
	
	public RemoteActor() {
		manager = Manager.getInstance();
	}
	
	public Manager getManager() {
		return (manager);
	}
	
	@Override
	public Receive createReceive() {
		return (receiveBuilder()
				.match( AddEventMessage.class, message -> handleMessage(message) )
				.match( GetEventsMessage.class, message -> handleMessage(message) )
				.build()
			   );
	}
	
	private void handleMessage(AddEventMessage message) {
		
	}
	
	private void handleMessage(GetEventsMessage message) {
		
	}
	
	public static Props props() {
		return ( Props.create(RemoteActor.class, () -> new RemoteActor()) );
	}
	
}